pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/templates/scripts/health-monitor.tpl.sh

```diff
@@ -13,7 +13,7 @@ function kubelet_monitoring {
   }
 
   function restart_kubelet {
-    pkill -f "kubelet"
+    pkill "kubelet"
```

Comment:
By default `pkill` will kill any process which has `kubelet` as a substring in its name. If I understand the issue correctly we want to kill process with exact match to name `kubelet` for that you can use following suggestion.
```suggestion
    pkill -x "kubelet"
```
From `pkill` manual
```
-x                Require an exact match of the process name, or argument list if -f is given.  The default is to match any substring.
```